src/main/kotlin/com/terraformation/backend/tracking/db/ObservationStore.kt

```diff
@@ -1374,13 +1374,17 @@ class ObservationStore(
             plantCountsBySpecies,
         )
       }
-      updateSpeciesTotalsTable(
-          OBSERVED_ZONE_SPECIES_TOTALS.PLANTING_ZONE_ID,
-          observationId,
-          plantingZoneId,
-          isPermanent,
-          plantCountsBySpecies,
-      )
+
+      if (plantingZoneId != null) {
+        updateSpeciesTotalsTable(
+            OBSERVED_ZONE_SPECIES_TOTALS.PLANTING_ZONE_ID,
+            observationId,
+            plantingZoneId,
+            isPermanent,
+            plantCountsBySpecies,
+        )
+      }
+
       updateSpeciesTotalsTable(
```

Comment:
Verifying the intent here: we want to update the site-level species totals based on ad-hoc observations?

Reply:
This is mostly due to how observation result store works. Since it relies on a `plantingSiteSpeciesMultiset` to get a lot of the data. 

I was thinking that, since ad-hoc plots are always temporary, it will not have permanent data so it should not be so consequential. But perhaps it is easier to write new queries to deal with ad-hoc plots on the result side.

Reply:
Observations of non-permanent plots don't count toward mortality rate calculations but they do still count toward the plant totals. If someone does an ad-hoc observation outside the site boundary, for example, does it make sense to include those plants in the site totals? The PRD says ad-hoc observations shouldn't be included in the plants dashboard, which suggests to me the answer is no.